### PR TITLE
add initial support for torch.ops.aten.neg.default converter

### PR DESCRIPTION
**Pull request by bowang007**
# Description

Initial support for a dynamo converter.

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/impl/unary/ops.py [hunk 1]
```diff
@@ -96,3 +96,19 @@ def sign(
...
+    source_ir: Optional[SourceIR],
+    name: str,
+    input_val: TRTTensor,
+) -> TRTTensor:
+    return convert_unary(
+        network,
+        target,
+        source_ir,
+        name,
+        trt.UnaryOperation.EXP,
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/conversion/impl/unary/ops.py:112]: Is this correct?

Comment (bowang007) [reply py/torch_tensorrt/dynamo/conversion/impl/unary/ops.py:112]: ohh it's a typo, let me update that.

File: tests/py/dynamo/converters/test_neg_aten.py
Comment (gs-olive) [inline tests/py/dynamo/converters/test_neg_aten.py:1]: Consider renaming this file `test_unary_aten.py`, so any subsequent unary op tests can go in the same file

File: tests/py/dynamo/converters/test_neg_aten.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from torch_tensorrt import Input
+
+
+class TestNegConverter(DispatchTestCase):
+    def test_neg(self):
+        class neg(nn.Module):
+            def forward(self, input):
+                return torch.neg(input)
+
+        inputs = [torch.randn(1, 10)]
```
Comment (gs-olive) [inline tests/py/dynamo/converters/test_neg_aten.py:15]: I believe `randn` defaults to `torch.int64`, so consider switching to `rand` for float or adding `.int()` for int, to the instantiation of the tensor

File: tests/py/dynamo/converters/test_neg_aten.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import torch
+import torch.nn as nn
+from parameterized import parameterized
+from torch.testing._internal.common_utils import run_tests
+from torch_tensorrt.dynamo.test_utils import DispatchTestCase
+from torch_tensorrt import Input
+
+
+class TestNegConverter(DispatchTestCase):
+    def test_neg(self):
```
Comment (gs-olive) [inline tests/py/dynamo/converters/test_neg_aten.py:10]: Consider adding cases to check both integer and float tensor negation. It could just be two functions in the same class.

**Review by gs-olive — COMMENTED** [review]
Changes look good to me, pending testing. Made some small comments on the test file.

File: tests/py/dynamo/converters/test_neg_aten.py (hunk 2 above)
Comment (bowang007) [reply tests/py/dynamo/converters/test_neg_aten.py:15]: Hey @gs-olive I went through the pytorch documentation here: https://pytorch.org/docs/stable/generated/torch.randn.html
I don't think it's `torch.int64` because in the examples we see float output tensor.

File: tests/py/dynamo/converters/test_neg_aten.py (hunk 3 above)
Comment (bowang007) [reply tests/py/dynamo/converters/test_neg_aten.py:10]: cool will do! thanks!

File: tests/py/dynamo/converters/test_neg_aten.py (hunk 2 above)
Comment (apbose) [reply tests/py/dynamo/converters/test_neg_aten.py:15]: I believe `randn` should cover the different data types. Also you could try including the `parameterized` test cases to check for different shapes of the input tensort.

File: py/torch_tensorrt/dynamo/test_utils.py
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/test_utils.py:1]: The changes here can be removed since it was added in #2246

**Review by gs-olive — COMMENTED** [review]
Just needs a rebase to the latest main, and should be good to go!

**Review by gs-olive — APPROVED** [review]
Looks good to me! Passes TorchScript CI (verified locally)

**Review by narendasan — APPROVED** [review]
LGTM
